### DISCUSSION TITLE
Feedback on message delivery

### DISCUSSION THREAD

**Discussion by JosephVoid** [capcom6/android-sms-gateway#37] [Q&A] [answered]
Hey, 

First of all, Thanks for making this project. Its really awesome, I would've contributed if I knew Kotlin : )

 I have been using this gateway with RabbitMQ & python, and it has been flawless so far. I was trying to see cases where the phone shuts down or suddenly there is no internet, and it appears that the cloud API return 'success' with pending status and then sends the SMS when its back on.
 
 I am curious how this works? Does the API store all of the messages in queue? and is me using a separate queue system redundant?
 
 Thanks

**Comment by capcom6** [comment, marked as answer]
Hello!

Thank you for your feedback!

When using the Cloud server, all messages are added to a queue, so a success response from the server only means that the request is valid and has been stored for delivery to the Android device. Therefore, the backend will return success even when the device is unavailable for any reason (such as being shut down, having no internet, etc.).

The use of a queue on the client side depends on your use case. For example, if you are sending messages from a background task, a queue might not be necessary for simple scenarios. However, if you are sending SMS as a result of a user action, it would be a good idea to handle it in the background (for instance, by using a queue) to improve user experience by minimizing the response time of your service.

I will consider adding some kind of device status endpoint in future versions to address cases where the device is unavailable. Currently, the only method is to check the message status. If it remains in the `Pending` state for more than 15 minutes (the minimum time for repeated background work in Android if the device didn't receive a push notification for some reason), this may indicate problems with the device.
